pydis_site/apps/home/apps.py

```diff
@@ -1,7 +1,38 @@
+from typing import Any, Dict
+
 from django.apps import AppConfig
 
 
 class HomeConfig(AppConfig):
     """Django AppConfig for the home app."""
 
-    name = 'home'
+    name = 'pydis_site.apps.home'
+    signal_listener = None
+
+    def ready(self) -> None:
+        """Run when the app has been loaded and is ready to serve requests."""
+        from pydis_site.apps.home.signals import SignalListener
+
+        self.signal_listener = SignalListener()
+        self.patch_allauth()
+
+    def patch_allauth(self) -> None:
+        """Monkey-patches Allauth classes so we never collect email addresses."""
+        # Imported here because we can't import it before our apps are loaded up
+        from allauth.socialaccount.providers.base import Provider
+
+        def extract_extra_data(_: Provider, data: Dict[str, Any]) -> Dict[str, Any]:
+            """
+            Extracts extra data for a SocialAccount provided by Allauth.
+
+            This is our version of this function that strips the email address from incoming extra
+            data. We do this so that we never have to store it.
+
+            This is monkey-patched because most OAuth providers - or at least the ones we care
+            about - all use the function from the base Provider class. This means we don't have
+            to make a new Django app for each one we want to work with.
+            """
+            data["email"] = ""
+            return data
+
+        Provider.extract_extra_data = extract_extra_data
```

Comment:
shouldn't this subclass, call super(), and then strip the email, in case allauth does something useful here?

Reply:
Allauth actually does nothing at all here, just `return data`. I figured this would be better for future work, since you can't actually tell Allauth which class to use - although I guess I could replace the entire class? I feel like that'd break things, though. I dunno.

What do you think?

Reply:
Yeah, that makes sense.